### PR TITLE
chore: WDS components in Table header

### PR DESCRIPTION
**Pull request by jsartisan**
Adds WDS components in Table Header

## Summary by CodeRabbit

- **New Features**
	- Introduced utility functions for handling table data, including downloading and transforming data into CSV or Excel formats.
	- Added a new `TableHeader` component for the `WDSTableWidget` with features like search, actions, adding new rows, and pagination.
- **Enhancements**
	- Enhanced feature flag checks for "ab_wds_enabled" and "release_anvil_enabled" to prioritize specific flags before general lookup.
	- Updated `ActionGroup` component to include a `size` property for the `IconButton`.
- **Refactor**
	- Removed redundant imports and renamed the `searchTableData` function to `onSearch` across multiple components to streamline search functionality.
	- Updated the `useListState` call to suppress text value warnings, improving code quality without altering functionality.

### CODE REVIEW

File: app/client/src/widgets/wds/WDSTableWidget/component/Table.tsx [hunk 1]
```diff
@@ -96,7 +95,7 @@ export interface TableProps {
     pageData: ReactTableRowType<Record<string, unknown>>[],
   ) => void;
   triggerRowSelection: boolean;
-  searchTableData: (searchKey: any) => void;
+  onSearch: (searchKey: any) => void;
```
Comment (jsartisan) [inline app/client/src/widgets/wds/WDSTableWidget/component/Table.tsx:98]: Just refactoring name of event.

File: app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/styles.module.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  & [data-table-header-pagination] > [data-field] {
+    width: 5ch;
+  }
+
+  & [data-table-header-pagination] > [data-field] input {
+    text-align: center;
+  }
+}
+
+@container table-header (inline-size < 600px) {
```
Comment (KelvinOm) [inline app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/styles.module.css:34]: Can we use `data-table-header-pagination` as container? This way we can adjust the pagination more precisely. So if the width of `data-table-header-pagination` has become too small, then we hide the text. Now it turns out that there is a space for the text (when the buttons are disabled), but we hide it.

File: app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/styles.module.css [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  & [data-table-header-pagination] > div {
```
[the remark this answers is not in the document]
Comment (jsartisan) [reply app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/styles.module.css]: ✅

File: app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/styles.module.css (hunk 2 above)
Comment (jsartisan) [reply app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/styles.module.css:34]: I tried the container query on the table pagination but for some reason, it was not working as expected. There was some overflow issue happening.

File: app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            {...rest}
+          />
+        </Flex>
+      </>
+    );
+  })();
+
+  return (
+    <>
+      <div className={styles["table-header"]} ref={tableHeaderRef}>
```
Comment (jsartisan) [inline app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/index.tsx:79]: @KelvinOm I have separated the container and layout element.

**Comment by jsartisan** [comment]
@KelvinOm removed feature flags

File: app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/styles.module.css [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+@container table-header (inline-size <= 320px) {
+  [data-layout] {
+    flex-wrap: wrap;
+  }
+
+  [data-layout] [data-field] {
+    width: 100%;
+  }
+
+  [data-layout] [data-table-actions] {
```
Comment (KelvinOm) [inline app/client/src/widgets/wds/WDSTableWidget/component/TableHeader/styles.module.css:57]: Seems like we don't need this. Allow the two components to take space depending on the width. 
```
  [data-layout] [data-table-actions] {
    flex: 0.5;
  }

  [data-layout] [data-table-header-pagination] {
    flex: 0.5;
  }
  ```
  
Apr-03-2024 14-41-43

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**